kotlinx-coroutines-core/common/src/CoroutineStart.kt

```diff
@@ -48,10 +48,29 @@ public enum class CoroutineStart {
      * Atomically (i.e., in a non-cancellable way) schedules the coroutine for execution according to its context.
      * This is similar to [DEFAULT], but the coroutine cannot be cancelled before it starts executing.
      *
+     * The coroutine stated with [ATOMIC] is guaranteed to start execution even if its [Job] was cancelled.
+     * This [CoroutineStart] option can be used to ensure resources' disposal in case of cancellation.
+     * For example, closing a channel used by a producer:
+     * ```
+     * fun CoroutineScope.producer(channel: SendChannel<Int>) =
+     *     launch(start = CoroutineStart.ATOMIC) {
+     *         try {
+     *             // produce elements
+     *         } finally {
+     *             channel.close()
+     *         }
+     *     }
+     * ```
+     *
+     * This is a **delicate** API. The coroutine starts execution even if its [Job] is canceled before starting.
```

Comment:
```suggestion
     * This is a **delicate** API. The coroutine starts execution even if its [Job] is cancelled before starting.
```
Both versions are correct, but we use "cancelled" throughout the codebase.